### PR TITLE
Fix 11184 by filtering the log

### PR DESCRIPTION
**Pull request by dzafman**
This is the first cut at this.  I don't have testing for it yet.

### CODE REVIEW

File: src/tools/ceph_objectstore_tool.cc [hunk 1]
```diff
@@ -1712,6 +1712,39 @@ int do_import_rados(string pool)
...
+      object_locator_t loc(i->soid);
+      // XXX: Do we need to set the hash? see also get_object()
+      // loc.hash = i->soid.get_hash();
+      pg_t raw_pgid = curmap.object_locator_to_pg(oid, loc);
+      pg_t pgid = curmap.raw_pg_to_pg(raw_pgid);
+
+      if (import_pgid.pgid == pgid) {
+        if (debug)
+          cerr << "Keep log entry for '" << i->soid << "'" << std::endl;
+        log.log.push_back(*i);
```
Comment (athanatos) [inline src/tools/ceph_objectstore_tool.cc]: For one thing, we need to keep the hitset objects.

Comment (athanatos) [inline src/tools/ceph_objectstore_tool.cc]: I'd also rather this were a method in PGLog.cc with a corresponding unit test.

Comment (dzafman) [inline src/tools/ceph_objectstore_tool.cc]: As far as hitset object entries we decided to not import any hitsets, see get_object().  So if we don't import them, they shouldn't be in the log either.  I think we assumed hitsets become out of date anyway.

**Comment by dzafman** [comment]
I'm going to push my latest without the final test code before we merge this.  I'll remove the DNM when it is there.

**Comment by dzafman** [comment]
Everything is the same here except I removed a commit to the tool's unit test.  Ready to merge.  Low risk.  Additional testing will come in later.

**Pull request merged**